protostar/testing/testing_summary.py

```diff
@@ -19,6 +19,12 @@
 from .testing_seed import Seed
 
 
+def calculate_skipped(
+    total_count: int, broken_count: int, failed_count: int, passed_count: int
+):
+    return total_count - (broken_count + failed_count + passed_count)
```

Comment:
`skipped_test_cases_count` and `skipped_test_suites_count` could be a `@property` of `TestingSummary`.

Reply:
They weren't before these changes, I'd address it in another PR.

Reply:
Why not in this PR?

Reply:
I would do this in this PR, it is small, related, change and good suggestion

Reply:
I don't think this is a super small, tiny change.

The problem here is that you want to make `skipped_tests` a part of the testing summary but we need a test collector result to calculate the skipped tests as well. That means that we can do one of these things:
- always calculate the skipped tests count when creating a test summary (which bonds the collector result existence with the testing summary creation in one scope)
- sometimes calculate the skipped tests count immediately when creating a testing summary and sometimes do this lazily
- always calculate the skipped tests count lazily (when needed). That may result in introducing some optionals/conditions, etc.

This PR is already pretty big and I'm afraid if we keep pumping up changes like that into it, it's going to turn unmanageable. Looks like it's about time to start decomposing work here. I didn't look that closely at the code, so I'm basically driven by my intuition here, nevertheless, I do not see any disadvantages in opening another PR for this. At most, we're going to have a tiny PR and just merge it fast.

I know it's 2 vs 1 but I have a pretty strong opinion that we should deal with this separately. If you insist, I can try to resolve this here.

Reply:
`TestingSummary` has `skipped` property that can be used to calculate skipped test cases count. To display skipped test suites count, I'd create a method in testing summary that accepts total test suites count because it contains most of the required information.

> Information expert (also expert or the expert principle) is a principle used to determine where to delegate responsibilities such as methods, computed fields, and so on.
Using the principle of information expert, a general approach to assigning responsibilities is to look at a given responsibility, determine the information needed to fulfill it, and then determine where that information is stored.
This will lead to placing the responsibility on the class with the most information required to fulfill it.
https://en.wikipedia.org/wiki/GRASP_(object-oriented_design)

Reply:
> `TestingSummary` has `skipped` property that can be used to calculate skipped test cases count.

I disagree. I looked at this and I think we have to rely strictly on the test collector result. If we use the `-x` flag, the tests are going to be omitted and therefore not added explicitly to the `skipped` property. We have to calculate the difference between cases collected by the collector and cases that have been executed in the current run. 

> To display skipped test suites count, I'd create a method in testing summary that accepts total test suites count because it contains most of the required information.

I added methods for obtaining the amount of both skipped test cases and skipped test suites.

I put up a separate PR for this as you didn't really respond to my arguments regarding splitting the work on this - #1315. I think we should merge this one and move on with any discussions about these skipped tests to a separate place - the new PR. The current one is very big and's been hanging for a while.

Reply:
Resolving, for further discussion, please post comments to #1315